simulation/eor_bubble_sim.pro

```diff
@@ -29,23 +27,83 @@ print, "Npix_selected: ", npix_sel
 
 ; Limit the range of frequencies in the uvf cube to the range of the obs
 freq_arr = (*obs.baseline_info).freq
-lim = minmax(freq_arr)
-freq_inds = where((freq_hpx GT lim[0]) and (freq_hpx LT lim[1]) )
-freq_hpx = freq_hpx[freq_inds]
+;lim = minmax(freq_arr)
+;freq_inds = where((freq_hpx GE lim[0]) and (freq_hpx LE lim[1]), count_inds )
+;if count_inds EQ 0 THEN  message, "No frequency overlap between shell and simulation"
+;freq_hpx = freq_hpx[freq_inds]
 nfreq_hpx = n_elements(freq_hpx)
 
+; making the coord array for h5s_select
+;coord_arr = array_indices(make_array(nfreq_hpx,npix_sel),findgen(nfreq_hpx*npix_sel))
+;coord_arr[1,*] = inds_select[coord_arr[1,*]]
+
 ;; Extract only these healpix indices from the file.
-;H5S_SELECT_ELEMENTS, dspace_id_eor, hpx_inds, /RESET
+; Somehow this ended up being slower than reading in the full array, so leave it out for now.
+;H5S_SELECT_ELEMENTS, dspace_id_eor, coord_arr, /RESET
 
 print, "Reading HDF5 file with EoR Healpix Cube"
 t0 = systime(/seconds)
 if n_elements(dat) eq 0 then dat = H5D_READ(dset_id_eor, FILE_SPACE=dpsace_id_eor)
 print, 'HDF5 reading time = ', systime(/seconds) - t0, ' seconds'
 
+if keyword_set(shellreplace) THEN BEGIN
+; Replace dat with a new array of the same variance and mean.
+    print, 'Replacing the shell'
+    sig = sqrt(variance(dat))
+    mu = mean(dat)
+    print, 'sig=', sig, 'mu=', mu
+    dat = randomn(seed, dims[1], dims[0])*sig + mu
+ENDIF
+
+; If l taper is set --- apply weights to l modes, up to the maximum l mode of the orthoslant map.
+; There isn't a standard way to do this strictly within IDL, so call healpy functions using IDL to Python bridge.
+if keyword_set(ltaper) THEN BEGIN
+    print, 'Applying l taper'
+    lmax_fhd = floor(180./obs.degpix)
+    lmax_hpx = 3*nside-1
+    if lmax_fhd GT lmax_hpx THEN BEGIN
+        print, "Input map resolution below FHD settings. lmax_fhd="+string(lmax_fhd)
+        lmax_fhd = lmax_hpx
+    ENDIF
+    hp = python.import('healpy')
```

Comment:
Please do another check to see if these functions exist for IDL because we currently don't have any python dependencies. If not, find out what the minimum healpy version is required for this to work and add a caveat to our dependencies section on the front-page readme, i.e. if generating visibilities from a EoR hd5 file using the simulation framework, then healpy version xx is required.

Reply:
It looks like IDL's healpix does provide a version of anafast which will compute a_lm, so it should be possible to do this without invoking the Python bridge. This would involve writing out a_lm components to a fits file and then reading them back in.

However, this was only a feature I added while trying to get the healpix binning right, and isn't essential to HEALPix-based EoR simulations. I could just cut it altogether.

Reply:
Gotcha. How about comment it out and put a comment explaining what the purpose is so if anyone needs to repeat this for different binning etc, they don't have to reinvent the wheel

Reply:
Okay! I've commented out the block and left an instruction in dictionary.md to uncomment to re-enable it.